### PR TITLE
Implement Index and a default file if directory given in import

### PR DESCRIPTION
**Pull request by john681611**
Ref to: https://github.com/sass/sass/pull/2456

I've not done Dart before so any help will be accepted.

- [x] Signed Contributors (waiting for legal to approve)
- [x] Rebase and remove extra commits

See sass/sass-spec#1206

### CODE REVIEW

File: test/node_api/importer_test.dart [hunk 1]
```diff
@@ -112,6 +112,18 @@ void main() {
...
+      var subDir = p.join(sandbox, 'dir');
+      await createDirectory(subDir);
+      await writeTextFile(p.join(subDir, 'index.scss'), 'a { b: c; }');
+      expect(
+          renderSync(new RenderOptions(
+              data: '@import "dir"',
+              includePaths: [sandbox],
+              importer: allowInterop(expectAsync2((_, __) {}, count: 0)))),
+          equalsIgnoringWhitespace('a { b: c; }'));
+    });
```
Comment (nex3) [inline test/node_api/importer_test.dart]: Index file support is more a part of the language than the API; it's an aspect of what it means for a filesystem import will mean. Which is to say, I think the sass-spec tests are sufficient and we don't need implementation-specific tests like this.

**Comment by nex3** [comment]
It looks like the CLA bot can't find your signature... is it associated with your GitHub account and the email you used for these commits?

**Comment by john681611** [comment]
Ah bugger I was advised to sign with my work email which is the secondary email in github

**Comment by nex3** [comment]
If you run `git --filter-branch --env-filter "GIT_AUTHOR_EMAIL=$WorkEmail; GIT_COMMITTER_EMAIL=$WorkEmail" origin/master; git push -f` that should update the email and make the CLA bot happy.

**Comment by john681611** [comment]
I signed it!

**Comment by nex3** [comment]
I'm still not seeing that email in the list of signers... I'm going to contact internal support to see if I can get this sorted out. Sorry for the trouble!

**Comment by nex3** [comment]
Sounds like we just have to wait a couple days for the lawyers to approve Marks & Spencer's paperwork.

**Comment by john681611** [comment]
No worries. Its the first time I or M&S have contributed to any big open source project AFAIK

**Comment by john681611** [comment]
I signed it!

Hoping it worked this time 🤞

**Comment by nex3** [comment]
Looks like not yet :confused:. I've poked the internal folks again.

**Comment by john681611** [comment]
Yeah, it keeps getting rejected saying our executive isn't high enough Head of Engineering just got rejected. We are working with our UK Google contact to figure out how high we have to go.

**Comment by john681611** [comment]
I signed it!

**Comment by nex3** [comment]
I *do* see the CLA internally now, but apparently googlebot still isn't happy? I'm not sure what's going on...

I'm sorry this process is being such a pain :sob:. I really appreciate you sticking it out.

**Comment by john681611** [comment]
Can you force skip the bot?

**Comment by nex3** [comment]
Unfortunately, I can't. I've reached out to the internal team again to see if there's something else you need to do.

**Comment by john681611** [comment]
I signed it!

**Comment by nex3** [comment]
celebrate

**Pull request merged**